beacon_node/execution_layer/src/lib.rs

```diff
@@ -1156,12 +1159,27 @@ impl<T: EthSpec> ExecutionLayer<T> {
                 &["new_payload", status.status.into()],
             );
         }
+        *self.inner.last_new_payload_errored.write().await = result.is_err();
 
         process_payload_status(execution_payload.block_hash(), result, self.log())
             .map_err(Box::new)
             .map_err(Error::EngineError)
     }
 
+    /// Update engine sync status.
+    ///
+    /// This will sometimes perform 2 upchecks, the 2nd one asynchronously.
+    pub async fn upcheck(&self) -> Result<(), Error> {
```

Comment:
Didn't understand why we cannot use `engine.upcheck()` directly instead of through the request method.

Reply:
Great point, thanks. I've simplified it to bypass `request(..)`